traits/ctraits.c

```diff
@@ -580,6 +580,39 @@ set_value ( PyObject ** field, PyObject * value ) {
     return 0;
 }
 
+static PyObject *
+get_trait_flag ( trait_object * trait, int mask ) {
+    PyObject * result;
+
+    if ( (trait->flags & mask) == 0 ) {
+        result = Py_False;
+    }
+    else {
+        result = Py_True;
+    }
+
+    Py_INCREF( result );
+    return result;
+}
+
+static int
+set_trait_flag ( trait_object * trait, int mask, PyObject * value ) {
+    int flag = PyObject_IsTrue(value);
+    if ( flag == -1 ) {
+        return -1;
+    }
+
+    if ( flag ) {
+        trait->flags |= mask;
```

Comment:
Side note: the use of bitwise operations on signed integers bugs me. Yes, I know it's fine, because Traits (or CPython, come to that), is unlikely to ever meet a machine whose signed integers don't use two's complement, but at some point we should change the flag type from `int` to `unsigned int`. Call it superstition.

Reply:
Opened #669 for this.